src/contracts/libraries/SlashingLib.sol

```diff
@@ -58,130 +56,91 @@ library SlashingLib {
-     * @param operatorMaxMagnitude The maxMagnitude of the operator for a given strategy
-     * @return bool true if either the operator or staker are fully slashed
-     */
-    function isFullySlashed(
-        StakerScalingFactors memory ssf,
-        uint64 operatorMaxMagnitude
-    ) internal pure returns (bool) {
-        return operatorMaxMagnitude == 0 || (ssf.getBeaconChainScalingFactor() == 0);
-    }
-
-    function getDepositScalingFactor(
-        StakerScalingFactors memory ssf
+    function scalingFactor(
+        DepositScalingFactor memory dsf
     ) internal pure returns (uint256) {
-        return ssf.depositScalingFactor == 0 ? WAD : ssf.depositScalingFactor;
-    }
-
-    function getBeaconChainScalingFactor(
-        StakerScalingFactors memory ssf
-    ) internal pure returns (uint64) {
-        return ssf.isBeaconChainScalingFactorSet ? ssf.beaconChainScalingFactor : WAD;
+        return dsf._scalingFactor == 0 ? WAD : dsf._scalingFactor;
     }
 
     function scaleSharesForQueuedWithdrawal(
         uint256 sharesToWithdraw,
-        StakerScalingFactors memory ssf,
-        uint64 operatorMagnitude
+        uint256 slashingFactor
     ) internal pure returns (uint256) {
-        /// forgefmt: disable-next-item
-        return sharesToWithdraw
-            .divWad(uint256(ssf.getBeaconChainScalingFactor()))
-            .divWad(uint256(operatorMagnitude));
+        if (slashingFactor == 0) {
+            return 0;
+        }
+
+        return sharesToWithdraw.divWad(slashingFactor);
     }
 
     function scaleSharesForCompleteWithdrawal(
         uint256 scaledShares,
-        StakerScalingFactors memory ssf,
-        uint64 operatorMagnitude
+        uint256 slashingFactor
     ) internal pure returns (uint256) {
-        /// forgefmt: disable-next-item
-        return scaledShares
-            .mulWad(uint256(ssf.getBeaconChainScalingFactor()))
-            .mulWad(uint256(operatorMagnitude));
-    }
-
-    function calcSlashedAmount(uint256 operatorShares, uint256 wadSlashed) internal pure returns (uint256) {
-        return operatorShares.mulWad(wadSlashed);
-    }
-
-    function decreaseBeaconChainScalingFactor(
-        StakerScalingFactors storage ssf,
-        uint64 proportionOfOldBalance
-    ) internal {
-        ssf.beaconChainScalingFactor = uint64(uint256(ssf.getBeaconChainScalingFactor()).mulWad(proportionOfOldBalance));
-        ssf.isBeaconChainScalingFactorSet = true;
+        return scaledShares.mulWad(slashingFactor);
     }
 
-    function updateDepositScalingFactor(
-        StakerScalingFactors storage ssf,
+    function update(
+        DepositScalingFactor storage dsf,
         uint256 existingDepositShares,
         uint256 addedShares,
-        uint64 maxMagnitude
+        uint256 slashingFactor
     ) internal {
         if (existingDepositShares == 0) {
             // if this is their first deposit for the operator, set the scaling factor to inverse of maxMagnitude
```

Comment:
Comment NIT: Inverse of `slashingFactor`. This isn't necessarily the inverse of the `maxMagnitude`. For Native ETH, it's the inverse of (mag*bcsf)